src/kioskboard.js

```diff
@@ -598,7 +598,13 @@
                   }
 
                   // add value by index
-                  theInputValArray.splice(theInputSelIndex, 0, keyValue);
+                  /** Prevents breaking on custom keys that insert multiple characters, for example, imagine a ".com" key, 
+                   * it would be inserted into the array as a single element and could no longer be deleted because the length of 
+                   * the input would be greater than the length of the array, adding Math.min (and comparing the respective lengths) 
+                   * to the backspace key prevents this problem but would delete the entire contents of the key (all characters, for example ".com") 
+                   * Instead, concatenate the main array of the contents with a possible array of keyValue, you can delete any single character.
+                   **/
+                  theInputValArray = theInputValArray.concat(keyValue.split(''));
```

Comment:
We need to use and update the selection index for adding each character because of the `allowRealKeyboard` option. 

So, my suggestion is:
- Spliting the `keyValue` for the possibility of the multiple characters.
- Creating a `for` loop for the new array of the `keyValue`.
- add, update, and dispatch events will be moved in this loop.

```js
var keyValArr = keyValue.split('').reverse();
for (var keyValIndex = 0; keyValIndex < keyValArr.length; keyValIndex++) {
	// add value by index
	theInputValArray.splice(theInputSelIndex, 0, keyValArr[keyValIndex]);

	// update input value
	input.value = theInputValArray.join('');

	// set next selection index
	input.setSelectionRange(theInputSelIndex + 1, theInputSelIndex + 1);

	// input trigger change event for update the value
	input.dispatchEvent(changeEvent);
}
```

What do you think?

Reply:
all ok for me 👍

Reply:
Do you want to replace the `keyElm.addEventListener('click', function (e) {...` code block with your current one and update this MR? (with discarding other ones)